File: python/array/leetcode/pascals_triangle.py
from typing import List


# 杨辉三角

class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        res = []
        for i in range(numRows):
            res.append([1] * (i + 1))
            for j in range(1, i):
                res[i][j] = res[i - 1][j - 1] + res[i - 1][j]
        return res

File: go/array/leetcode/pascals_triangle.go
package main

// 杨辉三角

func generate(numRows int) [][]int {
	var res [][]int
	for i := 0; i < numRows; i++ {
		v := make([]int, i+1)
		for j := range v {
			v[j] = 1
		}
		res = append(res, v)
		for j := 1; j < i; j++ {
			res[i][j] = res[i-1][j-1] + res[i-1][j]
		}
	}
	return res
}
